### PR TITLE
Removed opaque integer resources (deprecated in v1.8)

### LINKED ISSUE #55102: Remove support for opaque integer resources (deprecated in v1.8)
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

The opaque integer resources (OIR) feature was officially deprecated in v1.8,  v1.9 is the earliest possible version to remove it.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
**What this PR does / why we need it**:

* Remove opaque integer resources (OIR) support from the code base. This feature was deprecated in v1.8 and replaced by Extended Resources (ER).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #55102

**Release note**:

```release-note
Remove opaque integer resources (OIR) support (deprecated in v1.8.)
```

### CODE REVIEW

File: pkg/api/helper/helpers.go [hunk 1]
```diff
@@ -143,8 +143,7 @@ func IsStandardContainerResourceName(str string) bool {
 // IsExtendedResourceName returns true if the resource name is not in the
 // default namespace, or it has the opaque integer resource prefix.
```
Comment (dixudx) [inline pkg/api/helper/helpers.go]: remove OIR from the comment as well.

File: pkg/api/v1/helper/helpers.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 // default namespace, or it has the opaque integer resource prefix.
```
Comment (dixudx) [inline pkg/api/v1/helper/helpers.go]: Ditto.

File: pkg/api/v1/helper/helpers.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ import (
...
 func IsExtendedResourceName(name v1.ResourceName) bool {
 	// TODO: Remove OIR part following deprecation.
```
Comment (dixudx) [inline pkg/api/v1/helper/helpers.go]: Remove this TODO.

**Comment by ConnorDoyle** [comment]
ping
* @kubernetes/sig-scheduling-api-reviews 
* @kubernetes/sig-scheduling-pr-reviews 
* @vishh

**Comment by ConnorDoyle** [comment]
@davidopp could you please add the v1.9 milestone to this one?

File: pkg/apis/core/validation/validation_test.go [hunk 4]
```diff
@@ -5187,20 +5187,20 @@ func TestValidatePod(t *testing.T) {
...
 						Image:           "image",
 						ImagePullPolicy: "IfNotPresent",
 						Resources: core.ResourceRequirements{
 							Requests: core.ResourceList{
-								helper.OpaqueIntResourceName("A"): resource.MustParse("10"),
+								core.ResourceName("example.com/a"): resource.MustParse("10"),
 							},
 							Limits: core.ResourceList{
-								helper.OpaqueIntResourceName("A"): resource.MustParse("20"),
+								core.ResourceName("example.com/a"): resource.MustParse("10"),
```
Comment (zhangxiaoyu-zidif) [inline pkg/apis/core/validation/validation_test.go]: why change 20 to 10？

Comment (ConnorDoyle) [reply pkg/apis/core/validation/validation_test.go]: This test case is to verify the handling of valid extended resource requests. One difference between extended resources and opaque integer resources is that extended resources may not be overcommitted. So, if both request and limit are specified then they must be equal to pass validation.

Comment (ConnorDoyle) [reply pkg/apis/core/validation/validation_test.go]: We have other test cases in this file where extended resource request != limit, but we expect them to fail validation.

**Comment by ConnorDoyle** [comment]
ping @davidopp

**Review by vishh — APPROVED** [review]
/lgtm
/approve

File: test/e2e/scheduling/opaque_resource.go [hunk 5]
```diff
@@ -1,305 +0,0 @@
-/*
```
Comment (bsalamat) [inline test/e2e/scheduling/opaque_resource.go]: Can't this file be repurposed for testing extended resources?

Comment (ConnorDoyle) [reply test/e2e/scheduling/opaque_resource.go]: These tests always had flake problems and therefore are disabled by default. I agree we should have e2e tests for extended resources but given the state of these tests I’d prefer to add them back in another PR.

**Comment by ConnorDoyle** [comment]
_Attn:_ @kubernetes/api-approvers (@smarterclayton @thockin @lavalamp)

**Comment by ConnorDoyle** [comment]
```
ACK.  In progress
ETA: Unknown
Risks: Waiting for approver.
```

**Pull request merged**